mon-pix/tests/integration/components/signup-form-test.js

```diff
@@ -16,39 +16,60 @@ import {
 } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 import sinon from 'sinon';
+import ENV from '../../../config/environment';
 
 const FORM_CONTAINER = '.sign-form__container';
 const FORM_HEADER_CONTAINER = '.sign-form__header';
-const FORM_HEADER = '.sign-form-title';
-const EXPECTED_FORM_HEADER_CONTENT = 'Inscrivez-vous';
+const FORM_TITLE = '.sign-form-title';
 
 const INPUT_TEXT_FIELD = '.sign-form-body__input';
 const INPUT_TEXT_FIELD_CLASS_DEFAULT = 'form-textfield__input-container--default';
 
 const CHECKBOX_CGU_CONTAINER = '.signup-form__cgu-container';
 const CHECKBOX_CGU_INPUT = '#pix-cgu';
 const CHECKBOX_CGU_LABEL = '.signup-form__cgu-label';
-const UNCHECKED_CHECKBOX_CGU_ERROR = 'Veuillez accepter les conditions générales d\'utilisation (CGU) avant de créer un compte.';
 
 const CGU_LINK = '.signup-form__cgu .link';
-const CGU_LINK_CONTENT = 'conditions d\'utilisation de Pix';
 
 const SUBMIT_BUTTON_CONTAINER = '.sign-form-body__bottom-button';
 const SUBMIT_BUTTON = '.button';
-const SUBMIT_BUTTON_CONTENT = 'Je m\'inscris';
-
-const EMPTY_FIRSTNAME_ERROR_MESSAGE = 'Votre prénom n’est pas renseigné.';
-
-const EMPTY_LASTNAME_ERROR_MESSAGE = 'Votre nom n’est pas renseigné.';
-const EMPTY_EMAIL_ERROR_MESSAGE = 'Votre adresse e-mail n’est pas valide.';
-const INCORRECT_PASSWORD_FORMAT_ERROR_MESSAGE = 'Votre mot de passe doit contenir 8 caractères au minimum et comporter au moins une majuscule, une minuscule et un chiffre.';
 
 const userEmpty = EmberObject.create({});
 const CAPTCHA_CONTAINER = '.signup-form__captcha-container';
 
 describe('Integration | Component | signup form', function() {
 
   setupRenderingTest();
+  let intl;
+
+  beforeEach(function() {
+    intl = this.owner.lookup('service:intl');
+  });
+
+  describe('Localization', function() {
+    const originalLocale = ENV.APP.LOCALE;
+
+    afterEach(function() {
+      intl.setLocale(originalLocale);
+    });
+
+    [
+      { locale: 'fr-fr', expectedFormTitle: 'Inscrivez-vous' },
+      { locale: 'en-us', expectedFormTitle: 'Sign up' },
+    ].forEach(function(testCase) {
+      it(`${testCase.locale}`, async function() {
+        const expectedTitle = testCase.expectedFormTitle;
+        this.set('user', userEmpty);
+        intl.setLocale(testCase.locale);
+
+        // when
+        await render(hbs`{{signup-form user=user}}`);
+
+        // then
+        expect(find(FORM_TITLE).textContent).to.equal(expectedTitle);
+      });
+    });
+  });
```

Comment:
Pourquoi ne pas s'appuyer sur le test helper d'Ember Intl et mocker la traduction ?
https://ember-intl.github.io/ember-intl/docs/guide/testing

Reply:
Leur fonction `setupIntl(hooks)` utilise qunit derrière.  Ce n'est pas compatible avec nos tests actuels.

Reply:
Est-ce qu'il est possible de forker leur test helper pour en faire un custom qui tourne avec Mocha ?

Reply:
Ce que fait en gros le `setupIntl`:
- dans le `beforeEach`, il récupère le service `intl` et l'attache au `this` (pour pouvoir l'appeler ensuite depuis les tests)
- il spécifie une locale si besoin
- il rajoute des clés de traduction si besoin

Je ne pense pas que mocha puisse passer ses hooks en paramètres comme le fait `qunit`.

Reply:
Les fonctions de mocha (`beforeEach`, etc.) sont dans le namespace global donc j'ai l'impression que ça marcherait de faire un truc comme :
```js
function setupIntl() {
  beforeEach(function() { … });
}

describe('foo', function() {
  setupIntl();

  it('…', function() { … });
});
```

Reply:
le beforeEach est sorti dans une nouvelle fonction avec le dernier commit